File: packages/android-kotlin/src/main/kotlin/com/cinacoin/appkit/ui/CinacoinConnectButton.kt
package com.cinacoin.appkit.ui

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.AccountBalanceWallet
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.cinacoin.appkit.config.CinacoinTheme

/**
 * Cinacoin branded connect wallet button
 */
@Composable
fun CinacoinConnectButton(
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    style: ButtonStyle = ButtonStyle.Default,
    theme: CinacoinTheme = if (isSystemInDarkTheme()) CinacoinTheme.Dark else CinacoinTheme.Light
) {
    val backgroundColor = when (style) {
        ButtonStyle.Default -> theme.primary
        ButtonStyle.Outline, ButtonStyle.Minimal -> Color.Transparent
    }
    
    val contentColor = when (style) {
        ButtonStyle.Default -> Color.White
        ButtonStyle.Outline, ButtonStyle.Minimal -> theme.text
    }
    
    val border = when (style) {
        ButtonStyle.Outline -> BorderStroke(1.dp, theme.border)
        else -> null
    }
    
    Button(
        onClick = onClick,
        modifier = modifier
            .fillMaxWidth()
            .height(48.dp),
        colors = ButtonDefaults.buttonColors(
            containerColor = backgroundColor,
            contentColor = contentColor
        ),
        shape = RoundedCornerShape(12.dp),
        border = border,
        elevation = null
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.Center
        ) {
            Icon(
                imageVector = Icons.Default.AccountBalanceWallet,
                contentDescription = null,
                modifier = Modifier.size(20.dp)
            )
            Spacer(modifier = Modifier.width(8.dp))
            Text(
                text = "Connect Wallet",
                style = androidx.compose.material3.MaterialTheme.typography.labelLarge
            )
        }
    }
}

enum class ButtonStyle {
    Default, Outline, Minimal
}

@androidx.compose.runtime.Composable
private fun isSystemInDarkTheme(): Boolean {
    return androidx.compose.foundation.isSystemInDarkTheme()
}

File: packages/flutter-dart/lib/appkit_ui/cinacoin_connect_button.dart
import 'package:flutter/material.dart';
import '../appkit_config/cinacoin_theme.dart';

/// Cinacoin branded connect wallet button
class CinacoinConnectButton extends StatelessWidget {
  final VoidCallback onPressed;
  final CinacoinConnectButtonStyle style;
  final CinacoinTheme? theme;

  const CinacoinConnectButton({
    super.key,
    required this.onPressed,
    this.style = CinacoinConnectButtonStyle.defaultStyle,
    this.theme,
  });

  @override
  Widget build(BuildContext context) {
    final themeData = theme ?? _getThemeFromBrightness(context);
    
    Color backgroundColor;
    Color foregroundColor;
    BorderSide? border;

    switch (style) {
      case CinacoinConnectButtonStyle.defaultStyle:
        backgroundColor = themeData.primary;
        foregroundColor = Colors.white;
        border = null;
        break;
      case CinacoinConnectButtonStyle.outline:
        backgroundColor = Colors.transparent;
        foregroundColor = themeData.text;
        border = BorderSide(color: themeData.border);
        break;
      case CinacoinConnectButtonStyle.minimal:
        backgroundColor = Colors.transparent;
        foregroundColor = themeData.text;
        border = null;
        break;
    }

    return SizedBox(
      width: double.infinity,
      height: 48,
      child: ElevatedButton(
        onPressed: onPressed,
        style: ElevatedButton.styleFrom(
          backgroundColor: backgroundColor,
          foregroundColor: foregroundColor,
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(12),
            side: border ?? BorderSide.none,
          ),
          elevation: 0,
          padding: const EdgeInsets.symmetric(horizontal: 20),
        ),
        child: Row(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            const Icon(Icons.account_balance_wallet, size: 20),
            const SizedBox(width: 8),
            const Text(
              'Connect Wallet',
              style: TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.w600,
              ),
            ),
          ],
        ),
      ),
    );
  }

  CinacoinTheme _getThemeFromBrightness(BuildContext context) {
    final brightness = Theme.of(context).brightness;
    return brightness == Brightness.dark 
        ? CinacoinTheme.dark 
        : CinacoinTheme.light;
  }
}

enum CinacoinConnectButtonStyle {
  defaultStyle,
  outline,
  minimal,
}
